### ISSUE TITLE
Strange indentanzione problems

### ISSUE THREAD

**Issue by mirko-pagliai** [squizlabs/PHP_CodeSniffer#1134]
I'm using version 2.6.2 and are reported to me strange indentanzione problems.

First file:
http://pastebin.com/AT4jqpE8
Report:

```
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 56 | ERROR | [x] Line indented incorrectly; expected 8 spaces, found
    |       |     20 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
 67 | ERROR | [x] Line indented incorrectly; expected 8 spaces, found
    |       |     20 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
----------------------------------------------------------------------
```

Second file:
http://pastebin.com/G9TPC5GM
Report:

```
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 61 | ERROR | [x] Line indented incorrectly; expected 12 spaces, found
    |       |     20 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
 71 | ERROR | [x] Line indented incorrectly; expected 12 spaces, found
    |       |     20 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
----------------------------------------------------------------------
```

Third file:
http://pastebin.com/QEzqe0Eb
Report:

```
----------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 3 LINES
----------------------------------------------------------------------
 66 | ERROR | [x] Line indented incorrectly; expected 12 spaces, found
    |       |     20 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
 72 | ERROR | [x] Line indented incorrectly; expected 12 spaces, found
    |       |     20 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
 78 | ERROR | [x] Line indented incorrectly; expected 12 spaces, found
    |       |     20 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
----------------------------------------------------------------------
```

I have done a bit of testing, but nothing, I can not understand what the problem is

**Comment by aik099** [comment]
Potential duplicate of #1133.

**Comment by mirko-pagliai** [comment]
In the other discussion:

> This is because of scope (function/class) indentation. I guess you've enabled sniff, that wants class in a file to indented X nesting levels (e.g. 0)

But in my case, they are template files

**Comment by aik099** [comment]
If you look at example in that discussion they are also sort-of template files with `?>` and `<?php` used in the middle of PHP file, that causes issues.

**Comment by gsherwood** [comment]
That first example is best viewed with this structure:
```php
<div>
    <?php foreach ($banners as $banner) : ?>
        <div>
            <div>
                <?php
                    if ($foo) {
                    }
                ?>
            </div>
        </div>
    <?php endforeach; ?>
</div>
```
Here, the ScopeIndent sniff wants the IF statement indented to the same level as the PHP tags, but you've indented it one level more. The rest of the code in your example file is ignored because they are not scopes. They must be indented at least n spaces, but they can be indented more (as they are). But scope openers like IF must be indented to exactly the correct level.

Obviously, you prefer to indent your code one level from the tags, which this sniff doesn't support.

You're probably best off just excluding this sniff for your template files if you don't want to follow it's current rules.

Future dev could allow the sniff to be customised to allow for PHP tags to be treated like scope openers and force a single indent, but it's not currently possible. If that sounds like a feature you'd use, I can turn this issue into a feature request for that.
